app/ui/client/views/app/room.js

```diff
@@ -548,14 +578,35 @@ export const dropzoneEvents = {
 		e.stopPropagation();
 	},
 
-	'dropped .dropzone-overlay'(event, instance) {
+	async 'dropped .dropzone-overlay'(event, instance) {
 		event.currentTarget.parentNode.classList.remove('over');
 
 		const e = event.originalEvent || event;
 
 		e.stopPropagation();
 
-		const files = (e.dataTransfer != null ? e.dataTransfer.files : undefined) || [];
+		let files = (e.dataTransfer != null ? e.dataTransfer.files : undefined) || [];
+
+		if (files.length < 1) {
+			const transferData = e.dataTransfer.getData('text') !== '' ? e.dataTransfer.getData('text') : e.dataTransfer.getData('url');
+			if (e.dataTransfer.types.includes('text/uri-list')) {
+				const dropContext = $('<div>').append(e.dataTransfer.getData('text/html'));
```

Comment:
We've been avoiding to use jQuery whenever is possible to. Here, the vanilla alternative would be a [DocumentFragment](https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment) for `dropContext` and queries using the `querySelector()` API. However, jQuery safely handle nulls -- which allows you to write long chains of methods --, therefore you need to break the calls into smaller steps.